plugins/ssh-plugin/src/ssh-plugin-backend.ts

```diff
@@ -82,24 +83,57 @@ const getSshServiceName = async function (): Promise<string> {
     }
 };
 
+const updateConfig = function (host: string): void {
+    const sshDir = os.homedir() + '/.ssh';
+    const configFile = sshDir + '/config';
+    if (!fs.existsSync(configFile)) {
+        if (!fs.existsSync(sshDir)) {
+            fs.mkdirSync(sshDir);
+        }
+        fs.appendFileSync(configFile, '');
+        fs.chmodSync(configFile, '644');
+    }
+    const keyConfig = `\nHost ${host}\nHostName ${host}\nIdentityFile ${sshDir}/${host.replace('.', '_')}\n`;
+    const configContent = fs.readFileSync(configFile).toString();
+    if (configContent.indexOf(keyConfig) >= 0) {
+        const newConfigContent = configContent.replace(keyConfig, '');
+        fs.writeFileSync(configFile, newConfigContent);
+    } else {
+        fs.appendFileSync(configFile, keyConfig);
+    }
+};
+
+const writeKey = function (name: string, key: string): void {
+    const keyFile = os.homedir() + '/.ssh/' + name.replace('.', '_');
+    fs.appendFileSync(keyFile, key);
+    fs.chmodSync(keyFile, '600');
+};
+
 const generateKeyPair = async function (sshkeyManager: SshKeyManager): Promise<void> {
-    const keyName = await theia.window.showInputBox({ placeHolder: 'Please provide a key pair name' });
+    const keyName = await theia.window.showInputBox({ placeHolder: 'Please provide a key pair name (Host name e.g. github.com)' });
     const key = await sshkeyManager.generate(await getSshServiceName(), keyName ? keyName : '');
+    updateConfig(keyName);
+    writeKey(keyName, key.privateKey);
     const viewAction = 'View';
-    const action = await theia.window.showInformationMessage('Do you want to view the generated private key?', viewAction);
+    const action = await theia.window.showInformationMessage('Do you want to view the generated public key?', viewAction);
```

Comment:
We should inform the user that the generation of the key pair was successful.

Reply:
done